yew/src/virtual_dom/vtag.rs

```diff
@@ -110,6 +118,72 @@ impl Clone for VTag {
     }
 }
 
+#[cfg(feature = "ssr")]
+impl ToHtmlString for VTag {
+    fn to_html_string(&self) -> String {
+        let mut parts: Vec<String> = Vec::new();
+        let tag_name = htmlescape::encode_minimal(&self.tag).to_lowercase();
+
+        for c in tag_name.chars() {
+            let is_alnum = (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9');
+            if !is_alnum && c != '-' && c != '_' && c != ':' {
+                panic!("Invalid tag name: `{}`", &tag_name);
+            }
+        }
+
+        parts.push(format!("<{}", tag_name).to_string());
+
+        for (key_unclean, value) in &self.attributes {
+            let key = key_unclean.to_lowercase();
+            // checked, value (special if textarea), disabled, href?, selected,
+            // kind -> type if input, disallow ref, disallow LISTENER_SET, class
+
+            for c in key.chars() {
+                let is_alnum = (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9');
+                if !is_alnum && c != '-' && c != '_' && c != ':' {
+                    panic!("Invalid attribute name: `{}`", &key);
```

Comment:
Would it be better to have `ToHtmlString::to_html_string` return a `Result<String, Err>` instead?

Reply:
I've provided my suggestion for how this should be handled above.